src/coreclr/gc/gc.cpp

```diff
@@ -43719,7 +43719,7 @@ HRESULT GCHeap::StaticShutdown()
 
     for (int i = 0; i < gc_heap::n_heaps; i ++)
     {
-        delete gc_heap::g_heaps[i]->vm_heap;
+        // delete gc_heap::g_heaps[i]->vm_heap;
```

Comment:
I would just remove this line completely and put a comment in gcinterface.h. see below